### ISSUE TITLE
Feature of Interest

### ISSUE THREAD

**Issue by SiBell** [urbanobservatory/standards#8]
Is there benefit in deciding on a common set of [FeatureOfInterest](https://www.w3.org/TR/vocab-ssn/#SOSAFeatureOfInterest)'s.

I'd say keep these really generic, e.g. use the terms already on Newcastle's [data portal](http://newcastle.urbanobservatory.ac.uk/), such as _Air Quality_, _People_, _Weather_.

The combination of _FeatureOfInterest_ and _observedProperty_ potentially gives us a powerful way of filtering the data. I.e. "give me all observations with `FeatureOfInterest=weather` and `observedProperty=ThermodynamicTemperature`".

**Comment by lukeshope** [comment]
I agree. It also looks like you're only allowed the one `FeatureOfInterest`.

We might have to consolidate some of the categories on the Newcastle portal slightly. Perhaps...
* Air quality
* Mobility (people counts, vehicle counts, transport, etc.)
* Resources (electricity/gas consumption, sewage)
* Weather
* Water quality
* Ecology (bee hives, etc.)
* Earth (soil moisture, etc.)
* Noise

Not sure about 'earth' or 'noise'. Better ideas?

**Comment by SiBell** [comment]
The more thought I give this the harder it becomes!

Technically speaking _air quality_, _water quality_ and _noise_ may actually be _observedProperty_'s. Therefore is something like this better:

* Mobility (people counts, vehicle counts, transport, etc.)
* Utilities (electricity/gas consumption, sewage)
* Weather
* Hydrology (water quality, flow depth, flow rate) 
* Ecology (bee hives, etc.)
* Geology (soil moisture, etc.)
* Pollution (noise, air quality)

Although _Pollution_ isn't much better, too negative?

**Comment by aarepuu** [comment]
I had a think about this as well and these are my ideas:

- Mobility (people counts, vehicle counts, transport, etc.)
- Utilities (electricity/gas consumption, sewage)
- Atmosphere (noise, air quality)
- Biodiversity (bee hives, etc.)
- Geology (soil moisture, etc.)
- Freshwater (water quality, flow depth, water level, flow rate)
- Weather (_maybe even pot under atmosphere_)

What do you think?

**Comment by SiBell** [comment]
I'd be happy with weather going under atmosphere. I think we need to make sure that indoor atmosphere is in a different category to outdoor atmosphere though.

**Comment by SiBell** [comment]
Is it worth adding another _featureOfInterest_ for **diagnostics**? I.e. for observations of a device's battery voltage, or its RSSI (signal strength).

**Comment by EttoreHector** [comment]
I guess it really depends on who are / are going to be the main users of
our data platforms.
Such a feature of interest is mainly relevant to maintenance staff. For
what I can see there's nothing wrong having that feature been represented,
but I wonder whether we can think of any reasons why it shouldn't (maybe
security reasons of any kind..?)
In general I believe our data solution cannot be the perfect deal for all
possible scenarios of data use.

**Comment by SiBell** [comment]
It's a very good point. It almost makes the argument stronger for having a **diagnostics** _featureOfInterest_, at least at the database level, so that you can choose to filter out any observations of this type if the request has come from a member of the public.

**Comment by lukeshope** [comment]
I'm in favour of having something like that, though I would prefer to think of it as **condition** rather than **diagnostics**, if it's things such as signal strength, battery level, remaining rocket fuel, transmission rates, etc.

I think the security risks associated with RSSI or battery level are negligible, but there's no reason you can't restrict the observation and/or observation collection associated with the sensor if need be, while still describing the metric. You would just return a 401/403 status code for those parts of the graph.

**Comment by EttoreHector** [comment]
I am somehow still confused about the definition of FeatureOfInterest and ObservedProperty. If I stick with what I read in https://www.w3.org/TR/vocab-ssn/#SOSAFeatureOfInterest and https://www.w3.org/TR/vocab-ssn/#SOSAobservedProperty, I'd be tempted to say that 

* The level of NO2 is a FeatureOfInterest (and BTW, should I called it "NO2 level", "NO2 concentration" or just "NO2"?)

* "Concentration" is the ObservedProperty.

However, from what we discussed in this thread and from what I can see in Simon's examples (https://github.com/urbanobservatory/standards/tree/si-edits/api/ontology-examples) it should be more like:

* "Air Quality" being the FeatureOfInterest

* "NO2" (level, or concentration or with no qualifier) being the ObservedProperty

What would be the best way to go around?

**Comment by SiBell** [comment]
So my take on it was that many of the examples in the [SSN Docs](https://www.w3.org/TR/vocab-ssn/#SOSAFeatureOfInterest) the FeatureOfInterest is some physical thing e.g. an office, apartment, Atmosphere of Earth, a window, a tree, etc. In the interest of saving us from having to agree on dozens of different features of interest, we're veering towards some very top-level, broad features, e.g. 'utilities', 'atmosphere', 'hydrology', etc. I'd agree that 'Air Quality' isn't a good name for a FeatureOfInterest and 'Earth Atmosphere' is probably better for this.

The examples for observedProperty are things like CO2, temperature, battery, windSpeed, so I'd say that no2-concentration would fit in nicely with these and be a valid observedProperty. If you can have `{observedProperty: 'windSpeed', hasFeatureOfInterest: 'atmosphere'}`, i.e. the _speed_ of the _wind_ is an observed property of the _atmosphere_. Then I don't see why we couldn't have `{observedProperty: 'no2-concentration', hasFeatureOfInterest: 'atmosphere'}`, i.e. the _concentration_ of _NO2_ is an observed property of the _atmosphere_.

We could really do with deciding on the definitive list of Urban Observatory _FeatureOfInterest_'s and _ObservableProperty_'s soon so that we don't end up with one observatory using _no2-concentration_ and another _NO2level_.

**Comment by lukeshope** [comment]
I agree with what Si says. I think it's helpful with the `ObservedProperty` to make them as precise as possible, so for example *air temperature* rather than *temperature*, and *NO<sub>2</sub> concentration* rather than *NO<sub>2</sub>*. In other words, enough specificity in the name to be able to know if you're comparing apples and oranges.

I suppose *atmosphere* is probably sufficient for things like air quality, unless any of the UOs are planning on going extraterrestrial?

I suspect the `FeatureOfInterest` and `ObservableProperty` lists will need to grow organically, as we buy new sensors and integrate new systems. Perhaps we should pick an easy(ish) category to begin with (like meteorology) and then we expand on it through pull requests and discussion, when there are no obvious candidates?

**Comment by EttoreHector** [comment]
Thank you both.

**Comment by SiBell** [comment]
I agree it'll end up growing organically. Let's get an initial list up in the docs, and then, as you say, use pull requests and discussion to add more when we need to.

If we do go extraterrestrial then this would be the time to add a new feature of interest, e.g. _Space_.

I'd say _Meteorology_ is probably a bad name for a feature of interest given the early discussions about features of interest being more physical things. Therefore _atmosphere_ would be better.

Taking this into account and merging our suggestions above, should we start with the following:

- **atmosphere** (outdoor air-quality, outdoor air temperature, precipitation)
- **earth** (soil moisture, soil temperature)
- **infrastructure** (indoor air-quality, bridge vibration)
- **utilities** (electricity/gas consumption, flow rate of a sewage pipe)
- **transport** (vehicle counts, no. of bus passengers, train locations)
- **freshwater** (flow depth and flow rate of a river)
- **nature** (bee hive temperature, bird counts, pine tree locations)

N.b. happy to merge _utilities_ into _infrastructure_ for brevity.

**Comment by lukeshope** [comment]
Ah yeah, good point, bad example on my part.

I think all of the above makes sense for APIs where they only have sensors, and no contextual information about what those sensors relate in terms of rooms of a building or road junctions.

In some cases though, we will have a proper `featureOfInterest`. For example, a temperature sensor in a room might have a `platform` that is the wall (as represented in a BIM model for example), but the `featureOfInterest` would be the room (as represented in a BIM model as a space) rather than *infrastructure*.

If we have a "proper" `featureOfInterest`, should we attach the above "themes" using another relation? Either the sensor could be a member of a collection that represents the theme, or we could use [theme from DCAT](https://www.w3.org/TR/vocab-dcat-2/#Property:resource_theme).

The above all sounds like a sensible fallback for when there is no more precise `featureOfInterest`, or in the case of weather stations, where they're measuring the external environment.

**Comment by SiBell** [comment]
Totally see your point. A key factor is **scale**. My features of interest work well when looking at a city as a whole, asking questions like: "what's the average indoor air temperature across Birmingham", but less so for local scales: "is the average temperature of room G7 warmer than room G8 in the Urban Sciences building".

The way I've been handling this scale issue with what I've coded up so far is using nested platforms. So an example observation would be:

```
{
  resultTime: "2020-01-22T22:17:09.124Z",
  hasResult: {
    value: 22.6
  },
  madeBySensor: "thermistor-abc",
  observedProperty: "air-temperature",
  featureOfInterest: "infrastructure",
  isHostedBy: ["urban-sciences-building", "room-g7", "north-wall"]
}
```

The _urban-sciences-building_ is itself a _platform_, which hosts _room-g7_, which also hosts the _north-wall_ which hosts an air temperature sensor.

It's then really easy to query for any observations from the "urban-sciences-building", and just as easy to just get observations from the "north-wall".

It's a tricky one because the urban sciences building in this example is both a:
1. host for sensors, therefore fitting the definition of a _platform_.
2. a _feature of interest_ for anyone analysing the building.

**Comment by SiBell** [comment]
At some point soon I'll begin storing observations alongside a given _featureOfInterest_ and _observedProperty_, so I've had a very quick look around for any pre-existing lists of quantites that I could use as the _observedProperty_. I think I've hit the same problem that Ettore was alluding to earlier that existing quantities are very generic, e.g. "thermodynamic temperature" not "air temperature". E.g. [here](https://physics.nist.gov/cuu/Units/units.html) and [here](http://qudt.org/doc/2019/12/DOC_VOCAB-QUANTITY-KINDS-ALL-v2.1.html). 

If we go down the route of having a more specific _observedProperty_ values then it looks like we'll need to maintain our own list.

**Comment by SiBell** [comment]
If we used these [quantities](http://qudt.org/doc/2019/12/DOC_VOCAB-QUANTITY-KINDS-ALL-v2.1.html) as the _observedProperty_ and these [disciplines](http://qudt.org/doc/2019/10/DOC_VOCAB-DISCIPLINES-v2.1.html) as the _featureOfInterest_ does that work?

The issue I see is, for example, when you have a quantity of _Temperature_ and featureOfInterest of _Meteorology_ you have to rely on an assumption that the temperature must be of the air, e.g. rather than a ground temperature.

**Comment by SiBell** [comment]
So my previous suggestion was a terrible one, it doesn't take long to find an example that won't work. 

For example rain gauge accumulation observations would end up being something like `{"observedProperty": "Height", "featureOfInterest": "Meteorology"`, which doesn't even tell you it's rain!

I think we're best maintaining our own list, e.g. with observedProperty's of `rain-accumulation` and `rain-rate`, but can always use the QUDT quantities and disciplines as a starting point.

**Comment by SiBell** [comment]
It might be of interest to see the approach that this [paper](http://www.semantic-web-journal.net/system/files/swj2125.pdf) takes.

**Comment by SiBell** [comment]
Here's a whopping great big [list of Meteorological and Atmospheric Chemistry variables](http://cfconventions.org/Data/cf-standard-names/47/build/cf-standard-name-table.html) that could be used as ObservedProperty's.
